pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -129,10 +130,22 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 
 	var certChain, keyPEM, caBundle []byte
 	var err error
-	if features.PilotCertProvider == constants.CertProviderKubernetes {
+	if features.PilotCertProvider == constants.CertProviderKubernetes && signerName != "" && s.RA != nil {
+		log.Infof("Generating K8S-signed cert for %v using signer %v", names, signerName)
+		caCertFile := path.Join(ra.DefaultExtCACertDir, constants.CACertNamespaceConfigMapDataName)
+		if _, err := os.Stat(caCertFile); err != nil {
+			caCertFile = defaultCACertPath
+		}
+		certChain, keyPEM, _, err = chiron.GenKeyCertK8sCA(s.kubeClient.CertificatesV1beta1().CertificateSigningRequests(),
+			strings.Join(names, ","), hostnamePrefix+".csr.secret", namespace, caCertFile, signerName)
+		if err != nil {
+			return fmt.Errorf("failed genrating ker cert by k8s: %v", err)
+		}
+		caBundle = s.RA.GetCAKeyCertBundle().GetRootCertPem()
+	} else if features.PilotCertProvider == constants.CertProviderKubernetes {
 		log.Infof("Generating K8S-signed cert for %v", names)
 		certChain, keyPEM, _, err = chiron.GenKeyCertK8sCA(s.kubeClient.CertificatesV1beta1().CertificateSigningRequests(),
-			strings.Join(names, ","), hostnamePrefix+".csr.secret", namespace, defaultCACertPath)
+			strings.Join(names, ","), hostnamePrefix+".csr.secret", namespace, defaultCACertPath, "")
```

Comment:
Why no signerName here? Isn't that already a supported configuration?

Reply:
This is to keep our current behaviour, when we define `PILOT_CERT_PROVIDER` as `kubernetes` , Kubernetes will use `kubernetes.io/legacy-unknown` as the signer by default.